crates/ty_python_semantic/src/types/diagnostic.rs

```diff
@@ -1762,42 +1762,56 @@ pub(super) fn report_possibly_unbound_attribute(
     ));
 }
 
-pub(super) fn add_inferred_python_version_hint(db: &dyn Db, mut diagnostic: LintDiagnosticGuard) {
+/// Add a subdiagnostic to `diagnostic` that explains why a certain Python version was inferred.
+///
+/// ty can infer the Python version from various sources, such as command-line arguments,
+/// configuration files, or defaults.
+pub fn add_inferred_python_version_hint_to_diagnostic<D, A>(
```

Comment:
this function could plausibly be moved to the `ruff_db` crate now, since it's no longer specific to `ty_python_semantic` diagnostics. I'm happy to make that change, but it's a slightly bigger refactor -- the `PythonVersionSource` enum would also have to be moved to that crate